include/clock_control.h

```diff
@@ -72,6 +108,64 @@ static inline int clock_control_off(struct device *dev,
 	return api->off(dev, sys);
 }
 
+/**
+ * @brief Request clock to start with notification when clock has been started.
+ *
+ * User can request delayed start by providing exact information when clock
+ * should be ready. Driver ensures that clock is ready before requested time.
+ * It is the driver responsibility to take into account clock startup time.
+ * When clock is already running user callback will be called from the context
+ * of the function call else it is called from other context (e.g. clock
+ * interrupt).
+ *
+ * @param dev 	  Device.
+ * @param sys	  A pointer to an opaque data representing the sub-system.
+ * @param data	  Data structure containing a callback that is called when
+ *		  action is performed. MUST NOT BE ON STACK!. Can be NULL.
```

Comment:
Must not  be on stack or must be in persistent memory? How long the pointer must point to valid memory?

Reply:
Fixed the description.